### ISSUE TITLE
Commit 5a60935ba3367edc08aa5c52a2ab69914385e2d3 broke training for me

### ISSUE THREAD

**Issue by lowcaloriesyrup** [tdrussell/diffusion-pipe#347]
First off, I really appreciate your work! Thank you!

My dataset worked with wan2.1 and wanted to train against wan2.2. Updated to latest code and ran, but got the error below (always on step 31). Initially reverted back to 6940992455bb3bb2b88cd6e6c9463e7469929a70 per change logs in readme, but was able to bisect change sets until I got to 5a60935ba3367edc08aa5c52a2ab69914385e2d3, at which point training broke again. Each run used --regenerate_cache.

Error:
```
[2025-08-03 13:36:59,445] [INFO] [logging.py:107:log_dist] [Rank 0] step=31, skipped=0, lr=[2e-05], mom=[[0.9, 0.99]]
steps: 31 loss: 0.1636 iter time (s): 4.732 samples/sec: 0.211
/mnt/d/software_tools/diffusion-pipe/models/base.py:257: UserWarning: Using a target size (torch.Size([1, 16, 1, 180, 135])) that is different to the input size (torch.Size([1, 16, 1, 180, 134])). This will likely lead to incorrect results due to broadcasting. Please ensure they have the same size.
  loss = F.mse_loss(output, target, reduction='none')
```

dataset.toml (I know it's a bit of a mess, but it worked previously):
```
# Resolution settings.
# Can adjust this to 1024 for image training, especially on 24gb cards.
#resolutions = [1440]

#Aspect ratio bucketing settings
enable_ar_bucket = true
min_ar = 0.5
max_ar = 2.0
num_ar_buckets = 7

# Frame buckets (1 is for images)
frame_buckets = [1, 33]

[[directory]]
path = '/mnt/d/Projects/video-training/data/1440px/'
#resolutions=[1440]
size_buckets = [
    [768, 1024, 1],
    [1024, 1024, 1],
    [1080, 1440, 1],
    [1280, 704, 1],
    [1024, 768, 1],
]

# Frame buckets (1 is for images)
frame_buckets = [1]

# Reduce as necessary
num_repeats = 2
```

**Comment by tdrussell** [comment]
I confirmed this issue is caused by the manual size bucket [1080, 1440, 1], specifically the 1080 value because it isn't divisible by 16. It should be fixed for all models in the latest commit. Make sure to regenerate the cache when you run with the latest change.

Also, I think this failure would have always happened if you used that exact config. But if you changed the problematic size bucket, or used the auto bucketing, it wouldn't happen. It also only fails when it gets to the first image in the problematic size bucket, so you have to run it enough steps to eventually get it to fail. But anyway, it should be fixed now.

**Comment by lowcaloriesyrup** [comment]
I was using the exact dataset and exact dataset.toml (hence leaving it messy). Previous trainings ran for several thousand steps so it certainly would've hit the issue. Maybe something else was masking it?

I re-ran with latest commits (with --regenerate_cache) and am no longer seeing the issue.

Appreciate the quick fix!
